File: automobileinsuranceapi/src/main/java/com/hexaware/automobileinsurance/service/QuoteService.java
package com.hexaware.automobileinsurance.service;

import java.util.List;

import com.hexaware.automobileinsurance.dto.QuoteDTO;
import com.hexaware.automobileinsurance.model.Quote;

public interface QuoteService {

    Quote generateQuote(QuoteDTO dto);

    Quote getQuoteById(Integer quoteId);

    List<Quote> getAllQuotes();
    
    Quote calculatePremium(Integer proposalId);
}

File: automobile-insurance-frontend/src/services/QuoteService.js
import api from "../api/api";

const QuoteService = {

    getAllQuotes() {

        return api.get("/quotes");

    },

    getQuoteById(id) {

        return api.get(`/quotes/${id}`);

    },

    generateQuote(quote) {

        return api.post("/quotes", quote);

    },

    calculatePremium(proposalId) {

        return api.get(`/quotes/calculate/${proposalId}`);

    }

};

export default QuoteService;